core/arch/arm/mm/core_mmu.c

```diff
@@ -367,29 +367,43 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 
 	memory_map[last].type = MEM_AREA_NOTYPE;
 
+
+
+	/*
+	 * Assign region sizes, note that MEM_AREA_TEE_RAM always uses
+	 * SMALL_PAGE_SIZE if paging is enabled.
+	 */
+	for (map = memory_map; map->type != MEM_AREA_NOTYPE; map++) {
+		paddr_t mask = map->pa | map->size;
+
+		if (!(mask & CORE_MMU_PGDIR_MASK))
+			map->region_size = CORE_MMU_PGDIR_SIZE;
+		else if (!(mask & SMALL_PAGE_MASK))
+			map->region_size = SMALL_PAGE_SIZE;
+		else
+			panic("Impossible memory alignment");
+	}
+
 	/*
 	 * bootcfg_memory_map is sorted in order first by type and last by
 	 * address. This puts TEE_RAM first and TA_RAM second
 	 *
 	 */
-
 	map = memory_map;
 	assert(map->type == MEM_AREA_TEE_RAM);
 	map->va = map->pa;
 #ifdef CFG_WITH_PAGER
```

Comment:
Didn't get it
memory_map->region_size was already set to SMALL_PAGE_SIZE on line [382](https://github.com/OP-TEE/optee_os/pull/1252/commits/f7cbe03de3827762c1b9728e36cef76a4ace6e62#diff-21c5a372f38c092a429fd2b7e8f71ef3R382)
Or it's possible case when defined `CFG_WITH_PAGER` but `memory_map->region_size != SMALL_PAGE_SIZE` (regarless comment that `MEM_AREA_TEE_RAM always uses SMALL_PAGE_SIZE`)